virtio-devices/src/seccomp_filters.rs

```diff
@@ -85,6 +89,28 @@ fn virtio_blk_thread_rules() -> Result<Vec<SyscallRuleSet>, Error> {
     ])
 }
 
+fn virtio_blk_io_uring_thread_rules() -> Result<Vec<SyscallRuleSet>, Error> {
+    Ok(vec![
+        allow_syscall(libc::SYS_brk),
+        allow_syscall(libc::SYS_close),
+        allow_syscall(libc::SYS_dup),
+        allow_syscall(libc::SYS_epoll_create1),
+        allow_syscall(libc::SYS_epoll_ctl),
+        allow_syscall(libc::SYS_epoll_pwait),
+        #[cfg(target_arch = "x86_64")]
+        allow_syscall(libc::SYS_epoll_wait),
+        allow_syscall(libc::SYS_exit),
+        allow_syscall(libc::SYS_futex),
+        allow_syscall(SYS_IO_URING_ENTER),
+        allow_syscall(libc::SYS_madvise),
+        allow_syscall(libc::SYS_munmap),
+        allow_syscall(libc::SYS_read),
+        allow_syscall(libc::SYS_rt_sigprocmask),
+        allow_syscall(libc::SYS_sigaltstack),
+        allow_syscall(libc::SYS_write),
```

Comment:
The list looks good and I agree the only needed syscall from io_uring is `io_uring_enter(2)` for this specific thread. One question though, have you tested the io_uring setup? (Need a 5.6 host and enable io_uring through rust feature). I'm specifically asking about the testing because the CI does not validate anything regarding io_uring.

Reply:
@sboeuf Yes, I tested it on a VM with kernel `5.6.19`. For the experiments, I enforced the `io_uring` feature as default, and run the integration tests with our `dev_cli.sh` for both `gnu` and `musl` build. Besides some tests are failing because of OOM, I see many tests are passing fine, especially ones related to `blk`. I also did some negative tests (removing obvious required syscalls on the list) where I see the `audit` reporting missing syscalls for the `virtio_blk_io_uring` threads. Any other tests we should do, please let me know.

Reply:
That sounds like a good way to test things :+1: 
Let's merge it, we can still fix some details later if needed (once we move the CI to Ubuntu Groovy Gorilla host with 5.8 kernels).